tests/common/test_device_basics.hpp

```diff
@@ -97,6 +97,21 @@ class test_device_basics : public test_base<T> {
                   0.7f * algebra::getter::element(m2, i, j);
       }
     }
+
+    matrix_actor().set_zero(m2);
+    for (size_type i = 0; i < 6; ++i) {
+      for (size_type j = 0; j < 4; ++j) {
+        result += 0.4f * algebra::getter::element(m2, i, j);
+      }
+    }
+
+    matrix_actor().set_identity(m2);
+    for (size_type i = 0; i < 6; ++i) {
+      for (size_type j = 0; j < 4; ++j) {
+        result += 0.3f * algebra::getter::element(m2, i, j);
+      }
+    }
+
     return result;
```

Comment:
Sorry if I am not following, but is this variable adjusted somewhere to the new result?

Reply:
If I understand your question correctly, 
`test_device_basics` is actually run in both host and device code, and `result` from each architecture are compared to each other

Reply:
Ah, I see, then it makes sense